recipes/libpng/all/conanfile.py

```diff
@@ -81,8 +81,13 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, 'share'))
         tools.rmdir(os.path.join(self.package_folder, 'lib', 'libpng'))
         tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
+        self.copy("libpng_extra.cmake", dst=os.path.join(self.package_folder, "lib", "cmake", "libpng"))
 
     def package_info(self):
+        self.cpp_info.name = "libpng"
```

Comment:
Let's avoid setting cpp_info.name and just apply it to the relevant generators. This can be removed